### ISSUE TITLE
[Feature]: Self-update

### ISSUE THREAD

**Issue by EduardDurech** [termux/termux-app#2995]
### Feature description

An option to self-update Termux, can be over CLI in packages

### Additional information

Telegram and other apps with stand-alone apk's usually support this feature, it would require [`INSTALL_PACKAGES` permission](https://developer.android.com/reference/android/Manifest.permission#INSTALL_PACKAGES) and use the [`PackageInstaller`](https://developer.android.com/reference/android/content/pm/PackageInstaller)

**Comment by EduardDurech** [comment]
I am assuming this would require pre-built apk's to be hosted somewhere, but maybe someone more familiar with Android development supply more information

**Comment by Grimler91** [comment]
For installing latest version from f-droid you can use apkeep: 

```
apkeep --download-source f-droid --app com.termux@0.118.0 .
termux-open ./com.termux@0.118.0.apk
```

but it won't notify you about new available versions (could be scripted though), and it won't work with releases from github

**Comment by EduardDurech** [comment]
> but it won't notify you about new available versions (could be scripted though), and it won't work with releases from github

Yes, this was my intention though, as the F-Droid releases are less frequent, and it would be great to have the latest beta builds. I also only use F-Droid for Termux at the moment ;) It would also be nice to include self-updates for Termux:API, et cetera, for the same reason. Right now we have to build ourselves every time we'd like to include an un-released update

**Comment by tareksander** [comment]
Debug builds in Github Actions eventually expire, it would be nice if there was a dedicated repo to uploading the debug builds as releases (to not mess with f-droid picking up releases), then they won't expire. Once that's done making a script to get the releases and download the newest ones for each app would be easy.

**Comment by mtkleet** [comment]
https://github.com/ImranR98/Obtainium
Recently, I was looking for fdroid alternatibe and stumbled upon this. It's capable of updating Termux based on Github Actions. Offers much more, though.
